src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java

```diff
@@ -1479,4 +1490,40 @@ public GetClientStartTimeResult getClientStartTime(String clientKey) throws IOEx
       return GetClientStartTimeResult.newBuilder().build();
     }
   }
+
+  private void startPrometheusMetricsCollector() {
+    prometheusMetricsThread =
+      new Thread(
+        () -> {
+          while (true) {
+            try {
+              TimeUnit.SECONDS.sleep(30);
+              OperationsStatus operationsStatus = operationsStatus();
+              PrometheusPublisher.updateWorkerPoolSize(operationsStatus.getActiveWorkersCount());
+              PrometheusPublisher.updateDispatchedOperationsSize(operationsStatus.getDispatchedSize());
+              PrometheusPublisher.updatePreQueueSize(operationsStatus.getPrequeue().getSize());
+              PrometheusPublisher.updateClusterUtilization();
+              updateQueueSizes(operationsStatus.getOperationQueue().getProvisionsList());
+            } catch (InterruptedException e) {
+              Thread.currentThread().interrupt();
+              break;
+            } catch (Exception e) {
+              logger.log(Level.SEVERE, "Could not update RedisShardBackplane metrics", e);
+            }
+          }
+        },
+        "Prometheus Metrics Collector");
+
+    prometheusMetricsThread.start();
+  }
```

Comment:
how does prometheus work when people don't intend to use it? Does it break users anyway?  Not sure if there needs to be a config  `use_prometheus: true/false`

Reply:
Good point! It will send to the default port `8080` if port number is not set. I dont think this will break anything, but setting have a config explicitly might be useful to make everything more clear.

Reply:
You don't really have to use it - the metrics will be collected and exposed on default port 9090. I'm going to change it to explicitly configure the port or not start the web server by default though.